cli/lib/exec/run.js

```diff
@@ -96,58 +61,47 @@ const processRunOptions = (options = {}) => {
     options.key = util.getEnv('CYPRESS_RECORD_KEY')
   }
 
-  // if we have a key assume we're in record mode
-  if (options.key) {
-    args.push('--key', options.key)
-  }
-
-  if (options.outputPath) {
-    args.push('--output-path', options.outputPath)
-  }
-
-  if (options.parallel) {
-    args.push('--parallel')
-  }
-
-  if (options.port) {
-    args.push('--port', options.port)
-  }
-
-  if (options.quiet) {
-    args.push('--quiet')
-  }
-
-  // if record is defined and we're not
-  // already in ci mode, then send it up
-  if (options.record != null) {
-    args.push('--record', options.record)
-  }
-
-  // if we have a specific reporter push that into the args
-  if (options.reporter) {
-    args.push('--reporter', options.reporter)
-  }
-
-  // if we have a specific reporter push that into the args
-  if (options.reporterOptions) {
-    args.push('--reporter-options', options.reporterOptions)
-  }
-
-  // if we have specific spec(s) push that into the args
-  if (options.spec) {
-    args.push('--spec', options.spec)
-  }
-
-  if (options.tag) {
-    args.push('--tag', options.tag)
-  }
+  const optionList = [
+    { option: camelNames.autoCancelAfterFailures, condition: options.autoCancelAfterFailures !== undefined },
+    { option: camelNames.browser, condition: options.browser },
+    { option: camelNames.ciBuildId, condition: options.ciBuildId },
+    { option: camelNames.config, condition: options.config },
+    { option: camelNames.configFile, condition: options.configFile !== undefined, customProcessing: () => checkConfigFile(options) },
```

Comment:
I think it'd be worth moving this check specifically out of this helper and reducing this to:
```js
const runOpts = [
  'browser',
  'ciBuildId',
]

runOpts.forEach((opt) => P
  if (options[opt]) {
     args.push(cypressOptions[op].flag)
  })
})

```